gpt_json/types_streaming.py

```diff
@@ -0,0 +1,50 @@
+from enum import Enum
+from typing import Generic, Type, TypeVar
+
+from pydantic import BaseModel
+
+SchemaType = TypeVar("SchemaType", bound=BaseModel)
+
+class StreamEventEnum(Enum):
+    OBJECT_CREATED = "OBJECT_CREATED"
+    KEY_UPDATED = "KEY_UPDATED"
+    KEY_COMPLETED = "KEY_COMPLETED"
+    OTHER = "OTHER"
+
+
+class StreamingObject(Generic[SchemaType]):
+  schema_model: Type[SchemaType] = None
+
+  updated_key: str | None
+  value_change: str | None
+  event: StreamEventEnum
+  partial_obj: SchemaType
+
+  def __init__(self, obj_data: dict, prev_partial: 'StreamingObject[SchemaType]', proposed_event: StreamEventEnum, value_change: str | None = None) -> None:
+     super().__init__()
+     
+     # compute which key was most recently updated 
+     raw_recent_key = list(obj_data.keys())[-1] if obj_data else None
+     self.updated_key = raw_recent_key if raw_recent_key in self.schema_model.__fields__ else None
+     
+     self.event = proposed_event
+     if proposed_event == StreamEventEnum.KEY_UPDATED and self.updated_key is None:
+        # when there updated key is None, we haven't fully streamed the object's first key yet 
+        self.event = StreamEventEnum.OTHER
+     
+     # TODO: this is hacky. ideally we want pydantic to implement Partial[SchemaType]
+     # https://github.com/pydantic/pydantic/issues/1673
+     # my fix is to create the schema object with all string values for now
+     cleaned_obj_data = {field: "" for field in self.schema_model.__fields__.keys()}
```

Comment:
Any reason you can't just do the following to replace the dict comprehensions in 38 and 39:

```
cleaned_obj_data = {field: obj_data.get(field, "") for field in self.schema_model.__fields__.keys()}
```